baseplate.go

```diff
@@ -124,11 +149,14 @@ func Serve(ctx context.Context, server Server) error {
 			// Initialize a channel to pass the result of server.Close().
 			closeChannel := make(chan error)
 
-			// Tell the server to close.
+			// Tell the server and any provided closers to close.
 			//
 			// This is a blocking call, so it is called in a separate goroutine.
 			go func() {
-				closeChannel <- server.Close()
+				bc := batchcloser.New(preShutdown...)
```

Comment:
style: I would just use `args.preShutdown...` directly here rather than assigning it to a variable that is only used once.

```go
				bc.Add(args.postShutdown...)
```